cfht2caom2/tests/test_composable.py

```diff
@@ -79,37 +81,50 @@
 from cadcdata import FileInfo
 from caom2utils import data_util
 from caom2 import SimpleObservation, Algorithm, Instrument
+from caom2pipe import astro_composable as ac
 from caom2pipe import caom_composable as cc
 from caom2pipe import data_source_composable as dsc
 from caom2pipe import manage_composable as mc
 from cfht2caom2 import composable, cfht_name, metadata
-import test_main_app, cfht_mocks
+import test_main_app
+import cfht_mocks
 
 TEST_DIR = f'{test_main_app.TEST_DATA_DIR}/composable_test'
 
 
+@patch('caom2pipe.astro_composable.check_fits')
+@patch('caom2utils.data_util.get_local_headers_from_fits')
 @patch('cadcutils.net.ws.WsCapabilities.get_access_url')
-@patch('caom2pipe.execute_composable.CaomExecute._fits2caom2_cmd_local')
 @patch('caom2pipe.client_composable.CAOM2RepoClient')
-def test_run_by_builder(repo_mock, exec_mock, access_mock):
-    access_mock.return_value = 'https://localhost'
-    # should attempt to run LocalMetaCreate
-    repo_mock.return_value.read.side_effect = _mock_repo_read
-    repo_mock.return_value.create.side_effect = Mock()
-    repo_mock.return_value.update.side_effect = _mock_repo_update
-    getcwd_orig = os.getcwd
-    os.getcwd = Mock(return_value=TEST_DIR)
+def test_run_by_builder(
+    repo_mock, access_mock, util_headers_mock, fits_mock
+):
+    util_headers_mock.side_effect = ac.make_headers_from_file
+    # files are valid FITS
+    fits_mock.return_value = True
+
+    test_fqn = os.path.join(TEST_DIR, 'test_files')
+    if not os.path.exists(test_fqn):
+        os.mkdir(test_fqn)
     try:
-        # execution
-        test_result = composable._run_by_builder()
-        assert test_result == 0, 'wrong result'
+        access_mock.return_value = 'https://localhost'
+        # should attempt to run LocalMetaCreate
+        repo_mock.return_value.read.side_effect = _mock_repo_read
+        repo_mock.return_value.create.side_effect = Mock()
+        repo_mock.return_value.update.side_effect = _mock_repo_update
+        getcwd_orig = os.getcwd
```

Comment:
You could also use the `with mock.patch` construct to mock something at the block level.